internal/wazeroir/operations.go

```diff
@@ -948,19 +924,39 @@ func (b BranchTargetDrop) String() (ret string) {
 	return
 }
 
-// OperationUnreachable implements Operation.
+// UnionOperation implements Operation and is the compilation (engine.lowerIR) result of a wazeroir.Operation.
 //
-// This corresponds to wasm.OpcodeUnreachable.
+// Not all operations result in a UnionOperation, e.g. wazeroir.OperationI32ReinterpretFromF32, and some operations are
+// more complex than others, e.g. wazeroir.OperationBrTable.
 //
-// The engines are expected to exit the execution with wasmruntime.ErrRuntimeUnreachable error.
-type OperationUnreachable struct{}
+// Note: This is a form of union type as it can store fields needed for any operation. Hence, most fields are opaque and
+// only relevant when in context of its kind.
+type UnionOperation struct {
+	// OpKind determines how to interpret the other fields in this struct.
+	OpKind   OperationKind
+	B1, B2   byte
+	B3       bool
+	U1, U2   uint64
+	Us       []uint64
+	Rs       []*InclusiveRange
+	SourcePC uint64
```

Comment:
let's add this later as you suggested incrementally so that we can detect unnecessary portions! 
```suggestion
```

Reply:
oh you've deleted interpreterOp?

Reply:
yes! will do as I migrate more instructions

Reply:
👯